### PR TITLE
Support logos for courses (Model, helper and forms)

### PR DESCRIPTION
**Pull request by weiqingtoh**
Dependent on #765 - as I need some of the div classes to be specific in CSS classes. 

### Notes
 1. This PR adds support to upload logos for courses in forms. 
 1. Current logo is png, am getting Yanjie to do a svg version. Will swap that file out later on.
 1. *In addition, I've also changed the CSS variables convention to be more specific, ie. `$course-logo-form` rather than `$course-logo-small`. I feel that this will make customisation easier (all in 1 place, with more meaningful variable names), but let me know if this doesn't make sense. 

### After Action
After this, I will start implementing the Course logos on the sidebar, and course index page. 

#### Footnote.
This PR is part 1 of the course logo task in #624.

### CODE REVIEW

**Comment by lowjoel** [comment]
@weiqingtoh can you help rebase please?

File: app/assets/stylesheets/_variables.scss [hunk 1]
```diff
@@ -35,6 +35,6 @@ $picture-small: 75px !default;
 $picture-medium: 100px !default;
 
 //## Variables for icons or logos used in Coursemology.
-$course-achievement-badge-thumb:   $picture-thumb !default;
-$course-achievement-badge-small:   $picture-small !default;
-$course-achievement-badge-medium:  $picture-medium !default;
+$course-achievement-badge-form:   $picture-small !default;
```
Comment (lowjoel) [inline app/assets/stylesheets/_variables.scss]: So you're changing the scale sizes to actual places where it's used? That sounds like a good call.

File: app/assets/stylesheets/course/achievement.scss [hunk 2]
```diff
@@ -6,8 +6,8 @@
   .achievement-list,
   .achievement-form {
     .image > img {
```
Comment (lowjoel) [inline app/assets/stylesheets/course/achievement.scss]: on hindsight this should use some more class descriptors to reference the image, maybe .badge or something. thoughts?

File: app/views/course/admin/admin/index.html.slim [hunk 3]
```diff
@@ -6,6 +6,10 @@
     = f.input :description, label: t('course.courses.form.description_label'),
         placeholder: t('course.courses.form.description_placeholder')
 
+    = f.label t('course.courses.form.logo')
+    div.course-form = display_course_logo(current_course)
```
Comment (lowjoel) [inline app/views/course/admin/admin/index.html.slim]: ...this should be course-logo or something right?

File: spec/helpers/course/controller_helper_spec.rb [hunk 4]
```diff
@@ -136,5 +136,29 @@
...
+
+      context 'when a course logo is uploaded' do
+        let(:logo) { File.join(Rails.root, '/spec/fixtures/files/picture.jpg') }
+        before do
+          file = File.open(logo, 'rb')
+          course.logo = file
+          file.close
+        end
+
+        it 'displays the achievement badge' do
```
Comment (lowjoel) [inline spec/helpers/course/controller_helper_spec.rb]: achievement?

Comment (weiqingtoh) [inline spec/helpers/course/controller_helper_spec.rb]: Whoops .... copy paste errors >_<

File: app/assets/stylesheets/course/achievement.scss (hunk 2 above)
Comment (weiqingtoh) [inline app/assets/stylesheets/course/achievement.scss]: Not sure yet, but I think I will have a better answer when I start working on #761 - maybe we can defer this till then?

File: app/assets/stylesheets/_variables.scss [hunk 5]
```diff
@@ -35,6 +35,6 @@ $picture-small: 75px !default;
...
+
+$course-logo-form: $picture-small !default;
```
Comment (lowjoel) [inline app/assets/stylesheets/_variables.scss]: maybe course-admin-form-logo is better? idk

Comment (lowjoel) [inline app/assets/stylesheets/_variables.scss]: or maybe we should normalise it all at once... instead of in dribs and drabs

Comment (weiqingtoh) [inline app/assets/stylesheets/_variables.scss]: Do we have to worry about whether the variable name is too long? If not, course-admin-form-logo seems best, or even `course-admin-form-logo-height` and  `course-admin-form-logo-width` If we want to be specific.

Comment (lowjoel) [inline app/assets/stylesheets/_variables.scss]: no, prefer clarity over ambiguity. of course don't make the identifier 100 chars long...

**Pull request merged**